libs/sandbox-py/langchain_sandbox/tool.py

```diff
@@ -21,10 +24,35 @@ def _get_default_pyodide_sandbox() -> PyodideSandbox:
     )
 
 
-class PythonInputs(BaseModel):
-    """Python code to execute in the sandbox."""
+def _get_last_tool_artifact(
+    state: dict[str, Any],
+) -> dict[str, bytes | dict | None] | None:
+    if "messages" not in state:
+        error_msg = "Expected state to have 'messages' key."
+        raise ValueError(error_msg)
+
+    last_tool_message = next(
+        (
+            msg
+            for msg in reversed(state["messages"])
+            if isinstance(msg, ToolMessage) and msg.name == "python_code_sandbox"
+        ),
+        None,
+    )
+    if last_tool_message is None:
+        return None
+
+    if (
+        "session_bytes" not in last_tool_message.artifact
+        or "session_metadata" not in last_tool_message.artifact
+    ):
+        error_msg = (
+            "Expected tool artifact to have 'session_bytes' and "
+            "'session_metadata' keys."
+        )
+        raise ValueError(error_msg)
```

Comment:
Is langgraph ToolNode going to interpret this as a tool message by default?

Reply:
yes